# Assingment no.3

In [13]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import norm
from scipy.stats import chi2_contingency

In [14]:
df=pd.read_csv("Cutlets.csv")

In [15]:
df.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [16]:
df.shape

(35, 2)

In [20]:
sample1=pd.Series(df.iloc[ : ,0])
sample1

0     6.8090
1     6.4376
2     6.9157
3     7.3012
4     7.4488
5     7.3871
6     6.8755
7     7.0621
8     6.6840
9     6.8236
10    7.3930
11    7.5169
12    6.9246
13    6.9256
14    6.5797
15    6.8394
16    6.5970
17    7.2705
18    7.2828
19    7.3495
20    6.9438
21    7.1560
22    6.5341
23    7.2854
24    6.9952
25    6.8568
26    7.2163
27    6.6801
28    6.9431
29    7.0852
30    6.7794
31    7.2783
32    7.1561
33    7.3943
34    6.9405
Name: Unit A, dtype: float64

In [22]:
sample2=pd.Series(df.iloc[ : ,1])
sample2

0     6.7703
1     7.5093
2     6.7300
3     6.7878
4     7.1522
5     6.8110
6     7.2212
7     6.6606
8     7.2402
9     7.0503
10    6.8810
11    7.4059
12    6.7652
13    6.0380
14    7.1581
15    7.0240
16    6.6672
17    7.4314
18    7.3070
19    6.7478
20    6.8889
21    7.4220
22    6.5217
23    7.1688
24    6.7594
25    6.9399
26    7.0133
27    6.9182
28    6.3346
29    7.5459
30    7.0992
31    7.1180
32    6.6965
33    6.5780
34    7.3875
Name: Unit B, dtype: float64

In [23]:
p_value=stats.ttest_ind(sample1,sample2)
p_value

Ttest_indResult(statistic=0.7228688704678063, pvalue=0.4722394724599501)

here p_value=0.4722>0.05(alfa value)
hence we accept alternative hypothisis

# Q.2.Ans

In [33]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import norm
from scipy.stats import chi2_contingency

In [25]:
df1=pd.read_csv("LabTAT.csv")
df1

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0          185.35        165.53        176.70        166.13
1          170.49        185.91        198.45        160.79
2          192.77        194.92        201.23        185.18
3          177.33        183.00        199.61        176.42
4          193.41        169.57        204.63        152.60
..            ...           ...           ...           ...
115        178.49        170.66        193.80        172.68
116        176.08        183.98        215.25        177.64
117        202.48        174.54        203.99        170.27
118        182.40        197.18        194.52        150.87
119        182.09        215.17        221.49        162.21

[120 rows x 4 columns]

In [26]:
df1.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [27]:
df1.shape

(120, 4)

In [30]:
df1.describe()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
count    120.000000    120.000000    120.000000     120.00000
mean     178.361583    178.902917    199.913250     163.68275
std       13.173594     14.957114     16.539033      15.08508
min      138.300000    140.550000    159.690000     124.06000
25%      170.335000    168.025000    188.232500     154.05000
50%      178.530000    178.870000    199.805000     164.42500
75%      186.535000    189.112500    211.332500     172.88250
max      216.390000    217.860000    238.700000     205.18000

In [32]:
p_value=stats.f_oneway(df1.iloc[ : ,0],df1.iloc[ : ,1] ,df1.iloc[ : ,2],df1.iloc[:,3])
p_value

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

here p_value=2.1156e^57<0.05(alfa value)
hence we accept the null hypothisis

# Q .3. Ans

In [39]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import norm
from scipy.stats import chi2_contingency

In [40]:
df2=pd.read_csv("BuyerRatio.csv")
df2

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [41]:
df_A = df2.iloc[ : ,1:]
df_A

East  West  North  South
0    50   142    131     70
1   435  1523   1356    750

In [ ]:
# p_value=chi2_contingency(df_A)
p_value

here p_value=0.660309>apha value=0.05
hence we accept null hypothisis

# Q.4.ans

In [ ]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import norm
from scipy.stats import chi2_contingency

In [43]:
df4=pd.read_csv("Costomer+OrderForm.csv")
df4

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

In [45]:
df4.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [47]:
Phillippines=df4.Phillippines.value_counts()
Phillippines_Error_Free=Phillippines[0]
Phillippines_Defective=Phillippines[1]

In [48]:
Indonesia=df4.Indonesia.value_counts()
Indonesia_Error_Free=Indonesia[0]
Indonesia_Defective=Indonesia[1]

In [49]:
Malta=df4.Malta.value_counts()
PMalta_Error_Free=Malta[0]
Malta_Defective=Malta[1]

In [50]:
India=df4.India.value_counts()
India_Error_Free=Phillippines[0]
India_Defective=Phillippines[1]

In [51]:
df=[[Phillippines_Error_Free,Indonesia_Error_Free,PMalta_Error_Free,India_Error_Free],
    [Phillippines_Defective,Indonesia_Defective,Malta_Defective,India_Defective]]

In [52]:
df

[[271, 267, 269, 271], [29, 33, 31, 29]]

In [54]:
p_value=chi2_contingency(df)
p_value

(0.4014720642355302,
 0.9399382332004798,
 3,
 array([[269.5, 269.5, 269.5, 269.5],
        [ 30.5,  30.5,  30.5,  30.5]]))

p_value=0.9399>0.05(alfa.value)
hence we accept alternative hypothisis